File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0;
        int lth = nums.size() + 1;

        int a = 0;

        for(int i = 0; i < nums.size(); i++){
            sum += nums[i];
            while(sum >= target){
                lth = min(lth, i - a + 1);
                sum -= nums[a];
                a++;
            }
        }
        if(lth == nums.size() + 1){ return 0;}
        return lth;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        length = 10 ** 6
        amt = 0
        l = r = 0

        for i in range(len(nums)):
            amt += nums[i]

            while amt >= target:
                length = min(length, i - l + 1)
                amt -= nums[l]
                l += 1
                



        if length == 10**6:
            return 0
        return length
        